core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -341,7 +342,7 @@ public void cancel(Status status) {
         if (!internalCancel(Status.CANCELLED.withDescription("server cancelled stream"))) {
```

Comment:
I think this status here was intended to go to the client. The server could then just received the `status` as-is (in `internalCancel()`).